### DISCUSSION TITLE
Trakt watched sync with delete episode after x days

### DISCUSSION THREAD

**Discussion by Rouzax** [pymedusa/Medusa#3851] [Ideas & Feature requests]
I have numerous shows I watch were I have no desire to actual store the episodes after I've watched the episode.
It would be really helpful to have the ability to have Medusa import the watch status of episodes from Trakt and then on a show by show basis be able to set the delete after x days.
That way I can have shows where I want to delete the last watched episode after a week while other shows it could be after a day.

**Comment by fernandog** [comment]
Why not delete the episode in your MediaCenter after watch it?

**Comment by Rouzax** [comment]
That is a possibility but I don't like to give delete rights to Kodi. 
In my case my wife and kid also use Kodi and they are not that technical savvy.

**Comment by mrwsl** [comment]
This is quite a good feature request. Maybe add a label system to series to decide whether it is a show you want to archive or you want to watch and forget.

**Comment by p0psicles** [comment]
I've recently added a watched field to medusa db. So we can keep an administrative 'watched' status.
The idea was to be able to set this flag manually per ep. And also use our medusa kodi addon, to automate it. I think that in the future, we could tie actions like these to it.

So have trakt, update the 'watched' state. And then tie actions to it. like delete after x days.

On a side note, we first need to replace our current trakt library, in this PR https://github.com/pymedusa/Medusa/pull/4906. Before making any changes to the trakt integration.

**Comment by knowair** [comment]
+1 for importing episode's "Watched" status from Trakt

If this feature was to be added, could there be a secondary option to mark watched episodes as archived in medusa to stop the search for new improved quality?

i.e 
If episode marked as watched various options could be avialable
 1. Stop looking for better quality version. 
 2. Delete episode if airdate older than XX days/weeks/months
 3. Delete oldest if more than XX episodes in folder for this series
 4. All episodes watched AND series Status is Ended remove series from Medusa
